### PR TITLE
feat: creating shared components

### PR DESCRIPTION
**Pull request by iverbitskyi**
Created components for reuse for future improvement, refactored the code

### CODE REVIEW

**Review by Vlad-2109 — COMMENTED** [review]
I have some questions:

1) Liliia wrote a comment in dashboard for your ticket about sidebar. Will you add it later?
2) You wrote about schadcn/ui. It will be in other ticket or we won't use it?

**Comment by iverbitskyi** [comment]
> I have some questions:
> 
> 1. Liliia wrote a comment in dashboard for your ticket about sidebar. Will you add it later?
> 2. You wrote about schadcn/ui. It will be in other ticket or we won't use it?

File: src/shared/components/buttons/buttonUserPage.tsx
Comment (Virsta) [inline src/shared/components/buttons/buttonUserPage.tsx:1]: I suppose, you should combine buttonUserPage with submitButton  in one button with different options.

**Comment by iverbitskyi** [comment]
> I have some questions:
> 
> 1. Liliia wrote a comment in dashboard for your ticket about sidebar. Will you add it later?
> 2. You wrote about schadcn/ui. It will be in other ticket or we won't use it?

1. Yes, I will do it. I just thought it was in a separate ticket. But I see that it's probably better to do it in this one.
2. I realized that it makes no sense to use the shadcn library here. One of the reasons is that shadcn would preferably be used in conjunction with tailwindcss, which is not ok for us.

File: src/shared/components/buttons/buttonUserPage.tsx
Comment (iverbitskyi) [reply src/shared/components/buttons/buttonUserPage.tsx:1]: Okay, I'll fix it

**Comment by iverbitskyi** [comment]
https://github.com/VitalikMatvieiev/edu-platform/assets/58149045/96133296-168f-43a4-a6b3-57afa68aee93

File: src/shared/components/sidebar/_sidebar.scss [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+      border-radius: 25px;
+      box-shadow: 2px 2px 10px rgba(0, 0, 0, 0.2);
+      gap: 30px;
+      overflow: hidden;
+
+      &-icon {
+      }
+
+      &-text {
+      }
```
Comment (Vlad-2109) [inline src/shared/components/sidebar/_sidebar.scss:55]: Can you delete it, or in future we will have some styles for it?

File: src/pages/mainPage/mainPage.tsx [hunk 2]
```diff
@@ -1,18 +1,22 @@
...
-      {children}
-      <HeaderMain />
-      <CardOfCourse />
+    <div data-testid="main-page" className="main-page">
+      <Sidebar />
+      <div className="main-page-container">
+        <HeaderMain />
+        <CardOfCourse />
+        {children}
+      </div>
```
[the remark this answers is not in the document]
Comment (iverbitskyi) [reply src/pages/mainPage/mainPage.tsx:17]: Yes, I know about it. It's very cool. But there was no task to implement this within the framework of this ticket. I only needed to make a component for reuse. Later we will implement it.

[the remark this answers is not in the document]
Comment (iverbitskyi) [reply src/pages/mainPage/mainPage.tsx:17]: And here I inserted it just to test it

[the remark this answers is not in the document]
Comment (Vlad-2109) [reply src/pages/mainPage/mainPage.tsx:17]: It is unnecessary here, I will make it in my ticket.

File: src/shared/components/sidebar/sidebar.tsx [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+import { SidebarData, SidebarFooterData } from './sidebarData';
+import './_sidebar.scss';
+
+export const Sidebar = () => {
+  const [expanded, setExpanded] = useState(true);
+
+  return (
+    <aside className={`sidebar ${expanded ? 'expanded' : 'collapsed'}`}>
+      <button className="sidebar-button" onClick={() => setExpanded(!expanded)}>
```
[the remark this answers is not in the document]
Comment (iverbitskyi) [reply src/shared/components/sidebar/sidebar.tsx:16]: Ok, done, I fixed it

File: src/shared/components/sidebar/_sidebar.scss (hunk 1 above)
Comment (iverbitskyi) [reply src/shared/components/sidebar/_sidebar.scss]: Done

File: src/shared/components/sidebar/sidebar.tsx [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+      <ul className="sidebar-list">
+        {SidebarData.map((val, key) => {
+          return (
+            <Link to={val.link}>
+              <li key={key} className="sidebar-list-item">
+                <div>{val.icon}</div>
+                <div>{val.title}</div>
+              </li>
+            </Link>
```
[the remark this answers is not in the document]
Comment (iverbitskyi) [reply src/shared/components/sidebar/sidebar.tsx]: Done

File: src/shared/components/sidebar/sidebar.tsx [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+        <div className="sidebar-list-bottom">
+          {SidebarFooterData.map((val, key) => {
+            return (
+              <Link to={val.link}>
+                <li key={key} className="sidebar-list-item">
+                  <div>{val.icon}</div>
+                  <div>{val.title}</div>
+                </li>
+              </Link>
```
[the remark this answers is not in the document]
Comment (iverbitskyi) [reply src/shared/components/sidebar/sidebar.tsx]: Done

**Review by Vlad-2109 — DISMISSED** [review]

**Review by tanyaivattt — DISMISSED** [review]

**Review by tanyaivattt — APPROVED** [review]

**Review by Virsta — APPROVED** [review]

**Pull request merged**